dask/dataframe/core.py

```diff
@@ -1452,12 +1452,26 @@ def var(self, axis=None, skipna=True, ddof=1, split_every=False, dtype=None, out
             return handle_out(out, result)
         else:
             num = self._get_numeric_data()
-            x = 1.0 * num.sum(skipna=skipna, split_every=split_every)
-            x2 = 1.0 * (num ** 2).sum(skipna=skipna, split_every=split_every)
-            n = num.count(split_every=split_every)
-            name = self._token_prefix + 'var'
-            result = map_partitions(methods.var_aggregate, x2, x, n,
-                                    token=name, meta=meta, ddof=ddof)
+            values_dtype = num.values.dtype
+
+            if not np.issubdtype(values_dtype, np.number):
+                num.values.dtype = 'f8'
+
+            if skipna or skipna is None:
+                array_var = da.nanvar(num.values, axis=0, ddof=ddof, split_every=split_every)
+            else:
+                array_var = da.var(num.values, axis=0, ddof=ddof, split_every=split_every)
+
+            name = self._token_prefix + 'var--' + tokenize(num, split_every)
+
+            cols = num._meta.columns if isinstance(num._meta, pd.DataFrame) else None
```

Comment:
I think we use `is_dataframe_like(num._meta)` rather than explicit is instance checks.